### DISCUSSION TITLE
Custom build parameters for a single test branch build

### DISCUSSION THREAD

**Discussion by NickStull** [vercel/community#2740] [Help] [answered]
### Summary

My team is working on migrating from Create React App to NextJS. Rather than start a new Repo we put a top level directory called /Next and started building out the app there. 

Locally, this is easy. Just run `cd next` the `npm run dev` and it spins up the NextJS project. But now I would like to run a test deployment on what we have built to vercel. The problem is, I cannot for the life of me figure out how to change the presets to NextJS AND the root directory to '/next' for just a single test build on a single branch.

Any help would be greatly appreciated. Thank you for your time.

### Steps to Reproduce

All info is in the summary.

**Comment by amyegan** [comment, marked as answer]
You can override the build and development settings with a `vercel.json` config file on that branch. You can set the [`framework`](https://vercel.com/docs/concepts/projects/project-configuration#framework) to Next.js there. But Root Directory does not have a similar override.

For a single test build, it may be easiest to do a deployment from your local computer using [Vercel CLI](https://vercel.com/docs/cli). You could override the build settings on the branch, connect your `/next` directory to the project, and deploy from that directory.
